service/entityresolution/ldap/ldap_entity_resolution.go

```diff
@@ -0,0 +1,544 @@
+		slog.String("user_filter", c.UserFilter),
+		slog.String("email_filter", c.EmailFilter),
+		slog.String("client_id_filter", c.ClientIDFilter),
+		slog.Any("attribute_mapping", c.AttributeMapping),
+		slog.Bool("include_groups", c.IncludeGroups),
+		slog.Any("inferid", c.InferID),
+	)
+}
+
+// ResolveEntities implements the v1 ResolveEntities method
+func (s *LDAPEntityResolutionService) ResolveEntities(ctx context.Context, req *connect.Request[entityresolution.ResolveEntitiesRequest]) (*connect.Response[entityresolution.ResolveEntitiesResponse], error) {
+	ctx, span := s.Tracer.Start(ctx, "LDAP.ResolveEntities")
+	defer span.End()
+
+	resp, err := s.resolveEntities(ctx, req.Msg)
+	return connect.NewResponse(&resp), err
+}
+
+// CreateEntityChainFromJwt implements the v1 CreateEntityChainFromJwt method
+func (s *LDAPEntityResolutionService) CreateEntityChainFromJwt(ctx context.Context, req *connect.Request[entityresolution.CreateEntityChainFromJwtRequest]) (*connect.Response[entityresolution.CreateEntityChainFromJwtResponse], error) {
+	ctx, span := s.Tracer.Start(ctx, "LDAP.CreateEntityChainFromJwt")
+	defer span.End()
+
+	resp, err := s.createEntityChainFromJwt(ctx, req.Msg)
+	return connect.NewResponse(&resp), err
+}
+
+// resolveEntities handles the core entity resolution logic
+func (s *LDAPEntityResolutionService) resolveEntities(ctx context.Context, req *entityresolution.ResolveEntitiesRequest) (entityresolution.ResolveEntitiesResponse, error) {
+	connector, err := s.createConnector(ctx)
+	if err != nil {
+		return entityresolution.ResolveEntitiesResponse{}, 
+			connect.NewError(connect.CodeInternal, fmt.Errorf("failed to create LDAP connection: %w", err))
+	}
+	defer connector.close()
+
+	var resolvedEntities []*entityresolution.EntityRepresentation
+	
+	for idx, ident := range req.GetEntities() {
+		s.logger.Debug("resolving entity", "type", fmt.Sprintf("%T", ident.GetEntityType()), "id", ident.GetId())
+		
+		var entries []*ldap.Entry
+		var searchErr error
+
+		switch entityType := ident.GetEntityType().(type) {
+		case *authorization.Entity_ClientId:
+			entries, searchErr = connector.searchByClientID(ctx, entityType.ClientId)
+		case *authorization.Entity_EmailAddress:
+			entries, searchErr = connector.searchByEmail(ctx, entityType.EmailAddress)
+		case *authorization.Entity_UserName:
+			entries, searchErr = connector.searchByUsername(ctx, entityType.UserName)
+		default:
+			s.logger.Warn("unsupported entity type", "type", fmt.Sprintf("%T", entityType))
+			continue
+		}
+
+		if searchErr != nil {
+			s.logger.Error("LDAP search failed", "error", searchErr.Error())
+			return entityresolution.ResolveEntitiesResponse{}, 
+				connect.NewError(connect.CodeInternal, ErrGetRetrievalFailed)
+		}
+
+		var jsonEntities []*structpb.Struct
+		
+		if len(entries) == 0 {
+			// Handle entity not found - check if we should infer the identity
+			if s.shouldInferEntity(ident) {
+				entityStruct, err := s.entityToStructPb(ident)
```

Comment:
so its expecting the ldap search to return proto formatted entities? how does that work on the admin side? do they just provide a client filter that does conversion from whatever format they have to the json format of the client entity type?